### ISSUE TITLE
Allow passing custom flags / parameters / arguments to game executable

### ISSUE THREAD

**Issue by fasterthanlime** [itchio/itch#366]
Also environment variables

**Comment by HeatherSoron** [comment]
This is impacting recent Linux builds of a game in the itch.io refinery: it doesn't launch without the -force-glcore33 option, and there's currently no way to supply that flag from inside the itch client.

I might take a look at the relevant code, and see if I can implement it.

**Comment by fasterthanlime** [comment]
> I might take a look at the relevant code, and see if I can implement it.

that's generous, but there is no easy place for you to go ahead and add a text input field - haven't done the basics of "game-specific configuration dialog" yet so I wouldn't recommend spending too much time on that for now!

**Comment by HeatherSoron** [comment]
Gotcha, thanks for the heads-up on it being tricky. My current workaround is launching the game via shell script, so I'll stick with that for now.

**Comment by fasterthanlime** [comment]
> My current workaround is launching the game via shell script, so I'll stick with that for now.

Do you happen to be the developer of the affected game? Or just a savvy player?

**Comment by HeatherSoron** [comment]
> Or just a savvy player?

Savvy player, but I can forward info on to the developer (or point him here). If relevant, the game is 'hackmud', which was advertised on the refinery announcement page.

The developer sent me a link to issue #342 when I first reported the lack of launch flags. But, from poking around, it seems like this 'launch script' that's referenced might need to be developer-supplied, rather than player-supplied?

**Comment by fasterthanlime** [comment]
> from poking around, it seems like this 'launch script' that's referenced might need to be developer-supplied, rather than player-supplied?

Correct! @seanmakesgames will have to add a "launch.sh" with something like:

``` bash
#!/bin/bash
./Hackmud.x86_64 -force-glcore33 "$@"
```

However, I'm not sure the fix is required for everyone - it reminds me of a problem I've had on Linux with some OpenGL games: MESA will complain about OpenGL profile versions and GLSL shader versions unless you set `MESA_GL_VERSION_OVERRIDE=3.3`. However, when using optirun/primusrun, the nVidia drivers handled it just right.

**Comment by HeatherSoron** [comment]
Ah, so @seanmakesgames does have a github account. Wasn't sure.

> I'm not sure the fix is required for everyone - it reminds me of a problem I've had on Linux with some OpenGL games

That's exactly what I'm thinking, and part of why I think launch options might be preferable for hackmud, rather than a one-size-fits-all launcher script.

The hackmud client is programmed in Unity 5. Based on some [Reddit posts](https://www.reddit.com/r/KerbalSpaceProgram/comments/4h4doo/the_state_of_ksp_11_on_linux/) by the KSP community, it looks like Unity 5 is hit-or-miss with Linux, depending on hardware/driver details. On my machine (with an AMD hybrid GPU), hackmud requires `-force-glcore33`; but, I'm not sure how broadly that would apply to Linux users.

**Comment by seanmakesgames** [comment]
@fasterthanlime gonna jump in here and say that (and I think this is true) this is a low pri issue compared to getting more players into the game for @ethankaminski (e.g. the other stuff you are working on)
I think he just wanted to help out with itch dev and get his thing fixed. :)

**Comment by fasterthanlime** [comment]
@seanmakesgames no worries :) not prioritizing that over the other thing.

In the meantime, what you could do is ship a launch script, flag-free. Users like @ethankaminski could edit it by hand, it shouldn't be overwritten by updates as long as you don't change the launch script. Seems like an acceptable workaround?

**Comment by seanmakesgames** [comment]
Oh! Yeah, that's a great idea. Thanks @fasterthanlime :)

**Comment by seanmakesgames** [comment]
That will look like

```
#!/bin/bash
./hackmud_exe_name.x86_64 "$@"
```

and live in the root of the uploaded .zip, right?

**Comment by fasterthanlime** [comment]
> right?

correct!

**Comment by akien-mga** [comment]
I'd like this feature too, I recently wanted to workaround a bug in a Unity game by running it with `LC_ALL=en_US.UTF-8 %command%` on Linux (assuming that `%command%` would be replaced by the default launch command, à la Steam), but this doesn't seem possible yet.

**Comment by hagabaka** [comment]
Although there is currently no user interface for doing this, Windows users can accomplish this by creating a `.bat` file in the game installation directory, and Linux users by creating a shell script, which runs the executable with custom parameters and environment variables. It looks like Mac users have to create or edit an `.app` bundle. See https://github.com/itchio/dash/blob/master/configure.go#L475
